### PR TITLE
chore(atomic): migrate atomic-commerce-search-box-query-suggestions

### PR DESCRIPTION
**Pull request by alexprudhomme**
# AtomicCommerceSearchBoxQuerySuggestions Component Specification

## 📝 Requirements

_The `atomic-commerce-search-box-query-suggestions` component can be added as a child of an `atomic-search-box` component, allowing for the configuration of query suggestion behavior._

- [Highlighted suggestion]: When a query is present in the search box, the rest of the suggestion that is not matching the query should be highlighted.
- [Icon]: There should be a Search Icon whenever there are both recent queries and query suggestions. When there is only query suggestions, there should not be an icon.
- [Selecting the icon]: Should trigger a search.

## ✅ Checklist

- [x] 🧪 The component is [unit tested](https://docs.google.com/document/d/1lrsHAUROGMkRbsGlor02eYKF2mfrfBa0SOESxHFU3pI/edit?tab=t.0#heading=h.l9yzzrwf1i4k)
- [x] 🧪 The component includes [E2E tests](https://docs.google.com/document/d/1lrsHAUROGMkRbsGlor02eYKF2mfrfBa0SOESxHFU3pI/edit?tab=t.0#heading=h.z9tw07qe1lai)
- [x] 🗑️ Old Cypress tests exclusive to the component are removed
- [x] 📖 The component is documented in storybook with an `.mdx` file
- [x] ♿ The component complies with the [Web Content Accessibility Guidelines](https://www.w3.org/TR/WCAG21/).
- [x] 🌐 All strings intended for humans or assistive technology must be localized with i18n.
- [x] 📦 The Lit component is exported in the appropriate `index.ts` and `lazy-index.ts` files.
- [x] 🎨 CSS parts are documented still accessible.
- [x] 🦥 Slotted Content, public methods and properties are documented
- [x] 🔄 The component outputs the same Angular output as before with Stencil
- [x] 🏷️ The component declares the component type in the HTMLElementTagNameMap

https://coveord.atlassian.net/browse/KIT-3956

### CODE REVIEW

File: packages/atomic/src/components/commerce/atomic-commerce-search-box-query-suggestions/atomic-commerce-search-box-query-suggestions.spec.ts [hunk 1]
```diff
@@ -0,0 +1,243 @@
...
+      expect(
+        loadQuerySuggestActions(element.bindings.engine).fetchQuerySuggestions
+      ).toHaveBeenCalledWith(
+        expect.objectContaining({
+          id: element.bindings.id,
+        })
+      );
+    });
+
+    describe('renderItems', () => {
```
Comment (louis-bompart) [inline packages/atomic/src/components/commerce/atomic-commerce-search-box-query-suggestions/atomic-commerce-search-box-query-suggestions.spec.ts:123]: Those tests have the right `expectations/asserts` 👍

**Review by louis-bompart — COMMENTED** [review]
I left a "proposal" https://github.com/coveo/ui-kit/pull/5219 here.

I think a mix of both would be the right mix!

File: packages/atomic/src/components/commerce/atomic-commerce-search-box-query-suggestions/molecule-2.spec.ts [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+    it('should display an error component', async () => {});
+  });
+
+  describe('when inside a search box', () => {
+    it('should dispatch the search box suggestions event', async () => {});
+
+    it('should assign the bindings from the event', async () => {});
+
+    it('should call initialize after assigning bindings', async () => {});
```
Comment (louis-bompart) [inline packages/atomic/src/components/commerce/atomic-commerce-search-box-query-suggestions/molecule-2.spec.ts:38]: Here, the `initialize` method could be spied on & asserted on because it would be tested in `#initialize`

File: packages/atomic/src/components/commerce/atomic-commerce-search-box/e2e/atomic-commerce-search-box.e2e.ts [hunk 3]
```diff
@@ -50,7 +50,7 @@ test.describe('default', () => {
       });
 
       test('should update the searchbox input', async ({searchBox}) => {
-        await expect(searchBox.searchInput).toHaveValue(suggestionText);
+        await expect(searchBox.searchInput).toHaveValue(suggestionText.trim());
```
Comment (alexprudhomme) [inline packages/atomic/src/components/commerce/atomic-commerce-search-box/e2e/atomic-commerce-search-box.e2e.ts:53]: We now need to trim the suggestionText since lit adds html comments which turns into spaces so 

```
<span part="query-suggestion-text" class="line-clamp-2 break-all">
<!--?lit$255883212$-->recreational activities</span>
```

So we get errors like this in this test

```
    Error: Timed out 7000ms waiting for expect(locator).toHaveValue(expected)

    Locator: locator('atomic-commerce-search-box').getByPlaceholder('Search')
    - Expected string  - 4
    + Received string  + 1

    -
    -       
    -       query-suggestion-0
    + query-suggestion-0
    -    
```

**Review by dmgauthier — COMMENTED** [review]
Only a single import change for SVCC :heavy_check_mark:

**Review by dmgauthier — APPROVED** [review]

File: packages/atomic/src/components/commerce/atomic-commerce-search-box-query-suggestions/atomic-commerce-search-box-query-suggestions.spec.ts [hunk 4]
```diff
@@ -0,0 +1,262 @@
...
+      let object: SearchBoxSuggestions;
+      let items: SearchBoxSuggestionElement[];
+
+      beforeEach(async () => {
+        ({element} = await renderElements());
+        object = element.initialize();
+        items = object.renderItems();
+      });
+
+      it('should return the correct number of items', () => {
```
Comment (y-lakhdar) [inline packages/atomic/src/components/commerce/atomic-commerce-search-box-query-suggestions/atomic-commerce-search-box-query-suggestions.spec.ts:162]: Maybe also a test that handles empty suggestions

File: packages/atomic/src/components/common/suggestions/query-suggestions.spec.ts [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+  it('should render the icon if hasMultipleKindOfSuggestions is true', async () => {
+    const suggestion = await renderSuggestion({
+      hasMultipleKindOfSuggestions: true,
+    });
+
+    const icon = suggestion.querySelector('atomic-icon');
+
+    expect(icon).toBeInTheDocument();
+    expect(icon).toHaveAttribute('part', 'query-suggestion-icon');
+    expect(icon).toHaveAttribute('icon', 'icon');
```
Comment (y-lakhdar) [inline packages/atomic/src/components/common/suggestions/query-suggestions.spec.ts:84]: that's weird. what is the purpose of that attribute then?

File: packages/atomic/src/components/common/suggestions/query-suggestions.ts [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+};
+
+export interface RenderQuerySuggestionOptions {
+  icon: string;
+  hasQuery: boolean;
+  suggestion: Suggestion;
+  hasMultipleKindOfSuggestions: boolean;
+}
+
+export const renderQuerySuggestion = ({
```
Comment (y-lakhdar) [inline packages/atomic/src/components/common/suggestions/query-suggestions.ts:33]: I thought we were supposed to use the FunctionalCOmponent interface

File: packages/atomic/src/components/commerce/atomic-commerce-search-box-query-suggestions/atomic-commerce-search-box-query-suggestions.ts [hunk 7]
```diff
@@ -95,38 +104,38 @@ export class AtomicCommerceSearchBoxQuerySuggestions {
...
-      return (
-        <atomic-component-error
-          element={this.host}
-          error={this.error}
-        ></atomic-component-error>
-      );
-    }
+  @errorGuard()
+  render() {
+    return html``;
```
Comment (y-lakhdar) [inline packages/atomic/src/components/commerce/atomic-commerce-search-box-query-suggestions/atomic-commerce-search-box-query-suggestions.ts:133]:
```suggestion
    return html`${nothing}`;
```

**Review by y-lakhdar — APPROVED** [review]

File: packages/atomic/src/components/common/suggestions/query-suggestions.spec.ts (hunk 5 above)
Comment (alexprudhomme) [reply packages/atomic/src/components/common/suggestions/query-suggestions.spec.ts:84]: In my fixture, I put icon as 'icon' it was confusing to I changed it to 'fake-icon'.

File: packages/atomic/src/components/commerce/atomic-commerce-search-box-query-suggestions/atomic-commerce-search-box-query-suggestions.spec.ts (hunk 4 above)
Comment (alexprudhomme) [reply packages/atomic/src/components/commerce/atomic-commerce-search-box-query-suggestions/atomic-commerce-search-box-query-suggestions.spec.ts:162]: there is no logic when that happens, this would just test that slice and map work.

**Comment by alexprudhomme** [comment]
> #1. When `atomic-commerce-search-box` has `number-of-queries="3"` and `atomic-commerce-search-box-query-suggestions` has `max-with-query="5"` at the same time, the `number-of-queries > max-with-query`, it's better that we should add an explanation some where, so people will understand the situation better

I think this is out of scope for a migrating PR so I will log a separate ticket for this and we will go through it with the team.

> #2 . Accessibility : when searchbox has query suggestion, when you navigate through Clear button or Search button, the NVDA system still say Search field with suggestions. To begin navigating ....

I can log another ticket for this one as well.
